sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -920,4 +931,15 @@ object JdbcDialects {
  */
 private[spark] object NoopDialect extends JdbcDialect {
   override def canHandle(url : String): Boolean = true
+
+  /**
+   * Noop dialect always uses latest version
+   */
+  override def jdbcVersion(): Int = Int.MaxValue
```

Comment:
No op dialect can be used for some connectors that are not specialized (idk maybe Duck, SAP, Terraform), if we change something in base JDBC dialect, do we need versioning as well there?

Reply:
thats a good question. I intentionally set that these defaults always pick up latest version (by setting `Int.MaxValue`) but what you are saying makes sense as well